tests/modules/services/lieer/lieer-service.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  expectedService = pkgs.substituteAll {
```

Comment:
To avoid this `substituteAll` call and also to avoid having the tests depend on the actual `gmailieer` package you could leave the `lieer-service-expected.service` file as-is, remove this and instead add
```
nixpkgs.overlays = [
  (self: super: {
    gmailieer = super.gmailieer // { outPath = "@lieer@"; };
  })
];
```
to the `config` section below.

This overlay needs to be added to the other test as well to avoid the dependency. See the `i3-keybingings` test for an existing example.

Reply:
Added the overlay for the `lieer-service` test. The test in `programs/lieer.nix` doesn't depend on the package.

Reply:
Hmm, the module contains `home.packages = [ pkgs.gmailieer ]` so the test will definitely depend on the package. The PR test job is also red because the package is pulled in and fails to build on Darwin.